From: Julien Cristau <jcristau@debian.org>
Subject: Link against the libs we use

Avoids undefined references, ensures correct package dependencies.

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -27,6 +27,7 @@
 nv_drv_la_LTLIBRARIES = nv_drv.la
 nv_drv_la_LDFLAGS = -module -avoid-version
 nv_drv_ladir = @moduledir@/drivers
+nv_drv_la_LIBADD = $(PCIACCESS_LIBS) $(LIBDRM_LIBS) -lm
 
 nv_drv_la_SOURCES = $(nv_sources) $(riva_sources) $(g80_sources)
 
